src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -571,77 +659,127 @@ public static bool TryParse(string version, out SemanticVersion result)
 
         private static bool TryParseVersion(string version, ref VersionResult result)
         {
+            if (version.EndsWith('-') || version.EndsWith('+') || version.EndsWith('.'))
+            {
+                result.SetFailure(ParseFailureKind.FormatException);
+                return false;
+            }
+
+            string versionSansLabel = null;
+            var major=0;
+            var minor=0;
+            var patch=0;
+            string preLabel = null;
+            string buildLabel = null;
+
             var dashIndex = version.IndexOf('-');
+            var plusIndex = version.IndexOf('+');
 
-            // Empty label?
-            if (dashIndex == version.Length - 1)
+            if (dashIndex > plusIndex)
             {
-                result.SetFailure(ParseFailureKind.ArgumentException);
-                return false;
+                if (plusIndex == -1)
+                {
+                    // No buildLabel - buildLabel == null;
+                    preLabel = version.Substring(dashIndex+1);
+                    versionSansLabel = version.Substring(0, dashIndex);
+                }
+                else
+                {
+                    // No preLabel - preLabel == null;
```

Comment:
Can you add a comment that a hyphen character ('-') is allowed as part of the buildLabel (metadata) per the spec?  BTW why do you use "buildLabel" instead of "metadata" or "metadataLable" as referred to by the spec?

Reply:
Comments added and edited.